src/benchmark.jl

```diff
@@ -333,28 +346,36 @@ function WorkPrecisionSet(prob::AbstractRODEProblem,abstols,reltols,setups,test_
             dense_errors = dense_errors)
     end
   end
-  GC.gc()
   # Now time it
   for k in 1:N
     for j in 1:M
-      for i in 1:numruns
-        time_tmp[i] = @elapsed if !haskey(setups[k],:dts)
-          sol = solve(prob,setups[k][:alg];
-                kwargs...,
-                abstol=abstols[j],
-                reltol=reltols[j],
+      fails = 0
+      local t
+      @label START
+      try
+        t = if !haskey(setups[k],:dts)
+          @belapsed(solve($prob,$(setups[k][:alg]);
+                $kwargs...,
+                abstol=$(abstols[j]),
+                reltol=$(reltols[j]),
                 timeseries_errors=false,
-                dense_errors = false)
+                dense_errors = false), seconds = 0.1)
         else
-          sol = solve(prob,setups[k][:alg];
-                kwargs...,abstol=abstols[j],
-                reltol=reltols[j],dt=setups[k][:dts][j],
-                timeseries_errors=false,
-                dense_errors = false)
+          @belapsed(solve($prob,$(setups[k][:alg]);
+                          $kwargs...,abstol=$(abstols[j]),
+                          reltol=$(reltols[j]),dt=$(setups[k][:dts][j]),
+                          timeseries_errors=false,
+                          dense_errors = false), seconds = 0.1)
         end
+      catch
+        # sometimes BenchmarkTools errors with
+        # `ERROR: syntax: function argument and static parameter names must be distinct`
+        # so, we are catching that error and try a few times.
```

Comment:
why does it do that?

Reply:
I am not sure. It looks like something relate to macros.

Reply:
https://github.com/JuliaCI/BenchmarkTools.jl/issues/135 upstream issue